libraries/provider_database_mysql_user.rb

```diff
@@ -94,20 +94,42 @@ def ishash?
 
           db_name = new_resource.database_name ? "`#{new_resource.database_name}`" : '*'
           tbl_name = new_resource.table ? new_resource.table : '*'
+          test_table =  new_resource.database_name ? 'mysql.db' : 'mysql.user'
+          possible_privs = [
+            :select,
+            :insert,
+            :update,
+            :delete,
+            :create,
+            :drop,
+            :references,
+            :index,
+            :alter,
+            :create_tmp_table,
+            :lock_tables,
+            :create_view,
+            :show_view,
+            :create_routine,
+            :alter_routine,
+            :execute,
+            :event,
+            :trigger
```

Comment:
You are missing a few privileges here:
            :reload,
            :shutdown,
            :process,
            :file,
            :grant,
            :show_db,
            :super,
            :repl_slave,
            :repl_client,
            :create_user